### DISCUSSION TITLE
Turning in place

### DISCUSSION THREAD

**Discussion by rjbell4** [mjansen4857/pathplanner#279] [Q&A] [answered]
We are really loving PathPlanner, but there's one thing that eludes my understanding.  How can I make my (swerve) robot turn in place?  For example, suppose I wanted a simple path of going straight ahead, turn 90 degrees in place, and then go the "new" straight (i.e., 90 degrees perpendicular to the original direction of movement).  How I can represent this in PathPlanner?  I haven't been able to do anything that doesn't involve the robot doing a little "loop" as it turns.

**Comment by WarrenReynolds** [comment, marked as answer]
Its just not setup to do this (at the moment), without breaking it into three discrete operations. IE Finish following the first path, separate command to turn 90 degrees, then follow a new path.  However you can get very close to your requested behaviour by putting two waypoints very close together at the 90 degree corner with very low velocity overrides and then shorten the spline handles between the two points so they are almost touching the center point on the waypoint. NB: It gets a bit difficult to adjust things when everything is on top of each other.  The length of the tiny arc between the points and the velocity overrides are what determines how much time is given to complete the turn.

I'm pretty sure this feature is already on the wish list for next year.
